python/cudf/cudf/_lib/column.pyx

```diff
@@ -605,7 +609,9 @@ cdef class Column:
             else:
                 mask = as_buffer(
                     data=mask_ptr,
-                    size=bitmask_allocation_size_bytes(base_size),
+                    size=bitmask_allocation_size_bytes(
+                        max(size+offset, base_size)
+                    ),
```

Comment:
Having looked again, I think `size + offset` (rounded up to bitmask allocation size) is always the correct size for the null mask data pointer. Here is my rationale. `column_view.null_mask()` returns a `bitmask_type *` but without taking into account any `offset` the `column_view` may have. So if I have a `column_view` with size `10` and offset `200`, the bitmask pointer will point to a buffer of size `round_up(10 + 200)`, and the bits starting at `200` will be relevant for this view.

So, suggestion
```suggestion
                    size=bitmask_allocation_size_bytes(size+offset),
```